bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/ContentAssistRequest.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Dawid Pakuł and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pakuła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.ITextViewer;
+
+/**
+ * Describes the state that the content assistant is in when completing proposals.
+ * <p>
+ * Clients may use this class.
+ * </p>
+ *
+ * @since 3.24
+ * @see IContentAssistProcessorExtension2
+ * @noinstantiate This class is not intended to be instantiated by clients
+ */
+public final class ContentAssistRequest {
+	/**
+	 * Creates a new event.
+	 *
+	 * @param viewer text viewer
+	 * @param offset assist offset
+	 * @param autoActivated whether content assist was triggered by auto activation
+	 * @param incremental incremental flag {@link IContentAssistantExtension2}
+	 * @since 3.20
+	 */
+	public ContentAssistRequest(ITextViewer viewer, int offset, boolean autoActivated, boolean incremental) {
```

Comment:
If it's `@noinstantiate`, then can't we make the constructed package protected?
But thinking about it, some consumers may like to instantiate it for testing or to reuse the capability of contentAssistant in different context (eg to compute some quick-fixes). So the `@noinstantiate` seems a bit constraining and would maybe better removed.

Reply:
I'll transform to interface + private and/or inner class